client/app/services/session-manager.js

```diff
@@ -0,0 +1,68 @@
+import Ember from 'ember';
+import ENV from 'vilive/config/environment';
+
+const { service } = Ember.inject;
+
+/**
+ * makeRequest and authenticate should not be called
+ * outside this module, therfore I defined them here and
+ * didn't export them. Keeps things cleaner I think
+ */
+function makeRequest(data, path) {
+  return Ember.$.ajax({
+    url: ENV.yggdrasil.endpoint + path,
+    type: 'POST',
+    dataType: 'json',
+    data: JSON.stringify(data),
+    beforeSend(xhr) {
+      xhr.setRequestHeader('Accept', 'application/vnd.api+json');
+      xhr.setRequestHeader('Content-Type', 'application/vnd.api+json');
+    }
+  });
+}
+
+/**
+ * session is being explicitly passed in here
+ * because using service('session') directly
+ * doesn't jive well. It's an injected property
+ * and some ember majic is done.
+ *
+ * so left it on the service below and passed it in.
+ */
+function authenticate(session, user, path) {
+  const req = makeRequest(user, path); 
+  return session.authenticate('authenticator:yggdrasil', req);
+}
+
+export default Ember.Service.extend({
+  session: service('session'),
+
+  login(username, password) {
+    const user = {
+      data: {
+        type: "users",
+        attributes: {
+          username: username,
+          password: password
+        }
+      }
+    };
+
+    return authenticate(this.get('session'), user, ENV.yggdrasil.loginPath); 
```

Comment:
@copenhas I would like your opinion on this, not sure if I'm being extra cautious here. I defined `authenticate` and `makeRequest` without exporting them intentionally, because they should only be called internally. They are really just helper functions that I would not want a component to call directly.

It seemed reasonable, I did have to pass in the session like so, because I was not able to use it directly. `service('session')` returns and injected property, so you have to be able to call `get` from a ember object to resolve it. Not a huge deal to me, still seems cleaner than exposing authenticate/makeRequest for a component to call.